HarmonyOS/Day5 Assingment/Sanku Yogesh/App5/entry/src/main/java/com/example/myapplication/slice/SignupAbilitySlice.java

```diff
@@ -0,0 +1,172 @@
+import ohos.aafwk.ability.AbilitySlice;
+import ohos.aafwk.content.Intent;
+import ohos.agp.components.*;
+import ohos.agp.window.dialog.CommonDialog;
+import ohos.agp.window.dialog.IDialog;
+import ohos.agp.window.dialog.ToastDialog;
+import ohos.data.DatabaseHelper;
+import ohos.data.orm.OrmContext;
+import ohos.data.orm.OrmPredicates;
+
+import java.util.List;
+
+import static com.example.myapplication.utils.CheckingDataUtils.*;
+
+public class SignupAbilitySlice extends AbilitySlice {
+
+    private TextField firstName;
+    private TextField lastName;
+    private TextField password;
+    private TextField email;
+    private TextField mobile;
+    private Checkbox genderMale;
+    private Checkbox genderFemale;
+    private Button signupBtn2;
+    private Text firstNameText;
+    private Text lastNameText;
+    private Text emailText;
+    private Text emailExistsText;
+    private Text passwordText;
+    private Text mobileText;
+    private Text genderText;
+    private String firstNameString;
+    private String lastNameString;
+    private String passwordString;
+    private String emailString;
+    private String mobileString;
+    private String genderString;
+    OrmPredicates query;
+    DatabaseHelper helper;
+    OrmContext context;
+
+    @Override
+    protected void onStart(Intent intent) {
+        super.onStart(intent);
+        super.setUIContent(ResourceTable.Layout_ability_signup);
+
+        helper = new DatabaseHelper(SignupAbilitySlice.this);
+        context = helper.getOrmContext("UserInfo", "UserInfo.db", UserInfo.class);
+
+        firstName = (TextField) findComponentById(ResourceTable.Id_firstname_textField);
+        firstNameText = (Text) findComponentById(ResourceTable.Id_firstname_error);
+
+        lastName = (TextField) findComponentById(ResourceTable.Id_lastname_textField);
+        lastNameText = (Text) findComponentById(ResourceTable.Id_lastname_error);
+
+        password = (TextField) findComponentById(ResourceTable.Id_password_textField);
+        passwordText = (Text) findComponentById(ResourceTable.Id_password_error);
+
+        email = (TextField) findComponentById(ResourceTable.Id_email_textField);
+        emailText = (Text) findComponentById(ResourceTable.Id_email_error);
+        emailExistsText = (Text) findComponentById(ResourceTable.Id_email_exists_error);
+
+        mobile = (TextField) findComponentById(ResourceTable.Id_mobile_textField);
+        mobileText = (Text) findComponentById(ResourceTable.Id_mobile_error);
+
+        genderMale = (Checkbox) findComponentById(ResourceTable.Id_male_check_box);
+        genderFemale = (Checkbox) findComponentById(ResourceTable.Id_female_check_box);
+        genderText = (Text) findComponentById(ResourceTable.Id_gender_error);
+        genderString = "";
+
+        signupBtn2 = (Button) findComponentById(ResourceTable.Id_signup_button2);
+        signupBtn2.setClickedListener(new Component.ClickedListener() {
```

Comment:
Always try to use lambda in place of Anonymous class implementation. Please make this practice globally in all classes and projects.